### DISCUSSION TITLE
Node TRADING

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#2266] [Ideas]
**An idea to introduce a point-based system in which nodes (plots, trees, fruit patches etc.) are assigned points, to later trade with NPCs at Stone Haven**

Pre-requisite: Re-balance expansions so that everyone rolls the same amount of total points to one another. Example:

- Bob expands to 8th land, receives 4 trees (2 points each) and 2 plots (1 point each) = 10 points
- Jonny expands to 8th land, receives 1 gold node (5 points), 2 trees (2 points each) and 1 plot (1 points) = 10 points

Upon reaching level 20, players unlock access to stone haven where an NPC resides able to trade nodes with you.

The NPC allows players to swap their nodes for others, but at a cost. Using the example above, Let's say Jonny wants to trade his gold node (worth 5 points) for plots instead. Jonny would not be able to trade 1 gold for 5 plots, he would only receive 3 plots. Same sort of tax applies for all resources, these could be predetermined in a simple table. It can also be changed when seasons change, making it dynamic. (optional)

What does this achieve? 2 things:

1) Players can have what they want, to an extent. If someone does not want to farm gold or iron, they are free to trade them for plots or fruit patches.
2) The tax added when swapping these nodes creates a burn of sorts.. allowing players to swap 1:1 would be the simplest thing to do, but would also be lazy IMO and I feel as though you would need to create a drawback in swapping nodes.

"Won't everyone simply be swapping their iron/wood nodes for plots?" Well if they do, it means wood, stones, iron and gold are at a scarcity.. those farmers full of plots are not able to get wood etc. anymore and those other few who decided to specialise in iron/gold are now reaping the rewards with theorretically increased mineral prices (as majority only have plots). Players will swap TO what is profitable, which will keep burning nodes as they keep swapping... There could be interesting metas/markets around this. _(FYI Nodes would not be withdrawable)_

The more people/the more times nodes are traded, the rarer the nodes become as you get 80% of the 'points worth' back.

What I theorise would happen if implemented:

- Market prices would be changing more often
- Some resources would go through periods of scarcity due to how the players behave
- Players can decide to specialise in certain areas, less of x and more of z
- Freedom, but with a cost that hopefully helps to keep the game interesting

**ISSUE:**
The main issue I see is that players have already expanded and some people have a lot lower quantity of resources which means less points than others...Not really sure of the solution.
If it was up to me and only me, I would implement this, along with revamped skill trees, farm items (tractos, wheelbarrows, scarecrow I II etc.), new SFTS for low end players and just RELAUNCH SFL v4 refunding all resources back and start from 0. Also include tonnes of marketing this time..

^This is very far fetched but wanted to include it, not sure how else to incorporate a core mechanic change which would put people at a disadvantage as they already expanded with crap rolls.

Below is some crappy UI stuff i made to portray how it would look. **VALUES ARE RANDOM, NOT TO BE CRITICISED as they are not the main point (lol) of discussion and feedback requested**

![image](https://user-images.githubusercontent.com/108110065/219907785-8c3de267-8e6b-4468-8215-d3dd5d658649.png)

Appreciate any feedback at all. Disclaimer: My land is in top 50 for quantities, I am not biased and not crying for myself wanting more stuff, I have 43 plots to farm with and love it. BUT I do want a bit more fairness for people and want this issue to be sorted as best as possible. Paying for rerolls is not the solution IMO and I can explain that to people who are interested in why, but I said my bit in the following thread: https://github.com/sunflower-land/sunflower-land/discussions/2263#discussioncomment-5021333

Cheers cobbers

**Comment by future2back** [comment]
You kidding.

**Reply by TheDcol** [reply]
Hey mate, would appreciate further feedback, which part do you think i am kidding in and what would you propose instead? cheers

**Reply by VitorMaxtech** [reply]
Hey. As a player who got barely the minimum of the resourse nodes, holes and treepatches. I'm really sad about my land productivity over others. Example: I had 3 treepatches and when I expanded to the 10th expansion, I got no treepatches. And what makes me pretty sick about it is that the algorithm has a max given resourses at each expansion. that makes me think i'll be tha base no matter what i do or how much effort i give into my farm.
That IMAO is the point that sets players away, when they get to think what i just said. Also the best way to reach new players would be friendly indication. ppl don't feel safe to do that. 
my line of tought about it is that the system is kinda broken... Once the SFT have diferent weight over each farm. I'm talking about leveling.
The weight should be the same at each farm, and the diference should be at the diferent amount of diferent resourses. 
Everyone has the same 24hours at their own life, and game....
IMAO there is a higher chance of breaking the economy if the players don't believe the game.
I must also say that I love the game and playing it. And It would be much better if I felt not of this unbalancing without STFs.
thanks for reading..
good day everyone

**Reply by Xeko-92** [reply]
>Example: I had 3 treepatches and when I expanded to the 10th expansion, I got no treepatches.

Several things are wrong about this. Firstly, I believe you mean fruit patches, not trees. Secondly, the 10th expansion is guaranteed to get a fruit patch. Maybe you're talking about the 11th expansion, but there, you had 0% chance to get a fruit patch.

**Reply by VitorMaxtech** [reply]
You're right. *fruitpatches instead of treepatches
Aknowledged that 0% chance at the boulder expansion, wich is the 11th.
about the other things said?!

**Reply by TheDcol** [reply]
I don't disagree that the current system is not great :)

Point allocation expansions where everyone gets the same points but different resources would be better imo

ALong with 3 other proposals XD

**Reply by VitorMaxtech** [reply]
Point allocation expansions  where at each expansion, the algorithm evens the farms With that point system. IMO can work fine with some other side proposals

**Reply by VitorMaxtech** [reply]
11th expansion does not have a 0% chance fruitpatch. As I could se at least one farm with it.

**Comment by brian-man** [comment]
> The main issue I see is that players have already expanded and some people have a lot lower quantity of resources which means less points than others...Not really sure of the solution.

I already addressed this in my proposal in the other discussion.

**Reply by TheDcol** [reply]
Highest expansion land sets standard and allocate coupons for others to catch up?

Probably most realistic solution but there are going to be a lot more nodes around

**Reply by brian-man** [reply]
> Highest expansion land sets standard and allocate coupons for others to catch up?

I think any other approach means that whoever has the advantage today has that advantage permanently.

**Reply by TheDcol** [reply]
Fair point

Unless game was to completely reset 😅

**Comment by Xeko-92** [comment]
How do you think this would play out when a new NFT is introduced like the fruit NFTs? Would there be some sort of limitation in place during those times or would you just expect everyone to move everything over to fruit if something like that happened? Like let's say there's a boulder NFT that comes out once its utility is launched.

**Reply by TheDcol** [reply]
Thanks for the comment, good question!

I think it would be up to the player.. there would still be some minimum requirements but the player would decide whether to trade nodes everytime a new nft comes out to have an advantage. The issue with a player always trading nodes togain nft advantage is they would eventually run out of nodes due to the tax 😁.

**Comment by Chucrutty** [comment]
I don't really like this solution, my basic question here is what is the idea of ​​using Boulder in the game, will it be used to reset the lane or change nodes?
My proposal is that you have to reroll the entire farm, the specialization of plots nodes or anything else to me does not seem healthy for the game, imagine 1 player with FB and he makes an entire farm out of wood, or even other combos ....
what I find interesting and fun would be to have a reroll of the farm as a whole.... example farm lvl 1 - green grass
lvl2- blue grass (this happens when you decide to reset the farm) with this you would do a similar concept to your node score, but in my opinion it should be a score of how much resource you burned on the farm (wood, stone, iron, gold ) this would generate a score that would give you a bonus in the lvl 2 version of the farm - for example after resetting the farm you burn all the resources you used and you get a reset farm but with a general boost for example
 5%+yield
5% faster
With that, who has a good farm would have to decide to try to make a lvl 2 farm with a boost based on the resources you've already burned, or keep the farm as it is, so we fall into the gap I took a good farm but I would like the bonus of version 2 .... but since I'm satisfied with the lane I want to keep this one, or I didn't like the farm I'm going to burn it and make a lvl 2 farm with a bonus related to how many expansions you burned (sorry my english )

**Reply by TheDcol** [reply]
Thanks for the comment mate!

Regarding one of your comments: "1 player with FB making entire farm with wood..." They would have little to no plots, fruit patches and gold/iron. The main idea is the drawback, you can have lots of 1 thing if you want but barely anything or nothing in the rest. Encouraging trading resources between players

Regarding the farm reset you speak of, sounds like a prestige system, you should make a proposal for it and design it 😁 would be interesting to digest

**Reply by Chucrutty** [reply]
should i use Stone Haven for this proposal?

do you like this concept of prestige?

**Reply by TheDcol** [reply]
Absolutely go for it, nothing stopping you from creating a proposal

**Comment by Rook982** [comment]
I agree that this could be a viable idea if they won't go for a reset/reroll system. A resource swap system is nice. Ignoring the tax system for my suggestion for now cause I don't have a good alternative. Trading large resource nodes like fruit patches to stone,iron,gold there could exist a placement issue. An idea to fix this would be to make all stone, iron, gold and any future resource to take 2x2 spaces for easy swapping, if they want to swap to plots you now get 4 plots for that space.

**Comment by ShiningMinions** [comment]
@TheDcol I really appreciate you writing this proposal up :heart: . I think this mechanic could be interesting for other aspects of the game - however, with respect to resourceNodes, I am not in favor of this solution. Let me explain why:

> - Bob expands to 8th land, receives 4 trees (2 points each) and 2 plots (1 point each) = 10 points
> - Jonny expands to 8th land, receives 1 gold node (5 points), 2 trees (2 points each) and 1 plot (1 points) = 10 points

The main problem is that YOU are determining what each node is worth. That is not for you (or anyone else) to decide, that is for the free market to decide. Who knows if gold will be worth crap 2 years from now? In your above situation, Jonny receives a REALLY bad land, and the problem continues of what he perceives of "unfairness".

> The NPC allows players to swap their nodes for others, but at a cost. Using the example above, Let's say Jonny wants to trade his gold node (worth 5 points) for plots instead. Jonny would not be able to trade 1 gold for 5 plots, he would only receive 3 plots. Same sort of tax applies for all resources, these could be predetermined in a simple table. It can also be changed when seasons change, making it dynamic. (optional)

Okay, so let's say gold is worth a LOT in 2023. I'm going to sell all my nodes in exchange for goldNodes. Then in 2024, gold is worth crap, and now wood is the hot commodity - I will then trade all my gold nodes for trees. Then in 2025, gold is worth ALOT again, therefore I trade all my treeNodes for goldNode. 

Do you see the problem? My farm is slowly deteriorating, and my farm will eventually become blank. Couple this with me not being able to start a new farm (because of multi-accounting rule), leads to my eventual quitting.

As I mentioned in the other [node-balancing post](https://github.com/sunflower-land/sunflower-land/discussions/2263#discussioncomment-5026488), the main problem isn't with "balancing". The main issue is that I have 0 control getting the farm I actually want. My farm is forever dead-locked in this current system. I'm not allowed to start a new farm to "try again" (multi-account ban), I'm not able to trade my farmTiles for ones I actually want, and I am unable to re-roll my land.

**Reply by TheDcol** [reply]
Thanks Shining 💙

>The main problem is that YOU are determining what each node is worth. That is not for you (or anyone else) to decide, that is for the free market to decide.

This would be true if nodes were being traded between players, not between the player and the game (npc in this scenario). Also note, the points could be dynamic based on events or seasons. Trading nodes between players is something i am not considering due to the amount of abuse it would be subject to (multis creating new farms and the game still lacking solutions against them), but it is something I would not be totally against especially if the tax system was in place

> Do you see the problem? My farm is slowly deteriorating, and my farm will eventually become blank

So this is kind of the point, it discourages people switching every week chasing the hottest commodity of the day and is asking players 'what farmer will you be for the next while?', it also imo encourages players to further expand their lands and get more nodes if they are switching and burning a lot. Players can swap nilli willy but there are consequences if they keep switching for the hottest profit.

>The main issue is that I have 0 control getting the farm I actually want. My farm is forever dead-locked in this current system. I'm not allowed to start a new farm to "try again" 

I think sunflower isle was never initially intended for this, it was always spoken about the ability to expand forever and getting random lands. Not to say it will never be possible, just that maybe it was not the game's core idea

I do agree players should be given SOME freedom on how they want their land to look, but just stating that it was never a selling point for the game. I do not see devs just give players a blank canvas and say "go for it, build your land however you want to".

This node trading idea aims at allowing players to swap something they don't want for something they want, giving players SOME sort of freedom. Ultimately I want the system and experience to be better for everyone, these discussions can help sprout some interesting ideas so I appreciate the indepth conversation 💪☕

**Comment by brian-man** [comment]
The opening post offers some point values as a starting point for discussion:
> 1 - plot
> 2 - stone
> 3 - iron
> 5 - gold
> 2 - tree
> 3 - fruit patch
> 20 - boulder

Using this point system, I ran some numbers on the first 159315 lands using the visit json (2d old or less).

```
159315/159315

 1 |  5-13 points |   8 points credit max
 2 |  4- 5 points |   9 points credit max
 3 |  5- 7 points |  10 points credit max
 4 | 11-11 points |   4 points credit max
 5 |  3-15 points |  14 points credit max
 6 |  5-15 points |  22 points credit max
 7 |  4-15 points |  29 points credit max
 8 |  4-15 points |  37 points credit max
 9 |  4-18 points |  45 points credit max
10 |  7-16 points |  54 points credit max
11 |  7-17 points |  63 points credit max
12 |  8-24 points |  79 points credit max
13 |  7-24 points |  96 points credit max
14 | 25-53 points | 121 points credit max
15 |  2-19 points | 125 points credit max
16 |  4- 8 points | 128 points credit max
```
The "credit max" value indicates how many points to credit to farms that are below the best.

Considering the total points for the "best" farm is 275, a deficit of 128 points is kind of nuts.  It means the best farm is 87% better than the worst using this point system.

**Reply by Nimr1** [reply]
Well cant we then have those that have bad luck, buy stone rocks or gold rocks or fruit patches for a set price and then they will gradually become equal .

**Comment by ebuisan88** [comment]
I think a good implementation between RNG and the possibility to balance node is to have a TIER system.

For example we have a land expansion with the current tiers:
TIER 1: 2 plots 1stone 30%
TIER 2: 2plots 1 tree 1 stone 30%
TIER 3: 4 plots 20%
TIER 4: 4plots  1gold 10%

When you first expand your land you obtain a RNG TIER so the RNG aspect is still present. Then the user have the possibility to upgrade the TIER using resources, so if a player want to have the best tier he can choose to upgrade based on their preferences.

**Comment by TheDcol** [comment]
So a controlled reroll, where if you spend resources you are guaranteed to get better land? 

My one issue is that whales will just straight up have best lands with more nodes and hence able to mint more things. Not saying it's horrible, but it COULD be a point of contention

**Comment by Xeko-92** [comment]
The main thing I don't like is the node "burn". I think there should be a fee, but it should be an SFL fee or something.

**Reply by TheDcol** [reply]
As long as it prevents people swapping nodes too frequently to profit off it, i am open. 

Using your example, say i wanted to swap my gold node (5 points) for plots (1 point each), what do you base the cost off? 50 sfl per point? 5 gold per point? There are variety of options.

I mainly proposed burning node points because it would generate scarcity and try to conbat resource inflation, which i think is important for the longevity of the game rather than coming up with short term burns/mints all the time. It would not solve it but it would help

**Reply by Xeko-92** [reply]
The main issue I see with burning node points is what happens if you swap too much? You just have fewer nodes forever? You can't just start a new farm. It's way too punishing. There's no coming back from it really.

**Reply by TheDcol** [reply]
Yes that's the point, swapping should not be done too much. You swap resources you do not want and get ones you want to specialise into.

The issue of not having a drawback is what happens when a mint comes out requiring gold, everyone will swap to gold iron etc.. then when a mint or event comes out requiring crops everyone will swap for plots. There should be a cost to doing this and making players think for once on what they do in the game.

Or if you swap a lot and run out of nodes, you need to expand a lot to get more nodes.

A heavy sfl/resource cost would be an option too instead of node burning (burning points technically but ye)

**Reply by Xeko-92** [reply]
The thing is, there's a difference between "not done too much" and "never done". Burning the chance to ever get those points back would make people never want to do it. I'm basically okay with whatever SFL cost is decided upon, no matter how high. I just don't want to be in a position where you'll lose something forever. That just doesn't feel good.

**Reply by TheDcol** [reply]
Fair enough, i get that this might be too harsh for players
